File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        from collections import defaultdict, Counter
        st_dict = defaultdict(str)
        for i in range(len(s)):
            if st_dict[s[i]]:
                if st_dict[s[i]] != t[i]:
                    return False
            else:
                st_dict[s[i]] = t[i]
        dup_check = [] 
        for key in st_dict:
            if st_dict[key] in dup_check:
                return False
            else:
                dup_check.append(st_dict[key])

        return True

File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char,char> umap;
        unordered_map<char,char> reverse;
        for (int i = 0; i < s.length(); i++) {
            if (umap.find(s[i]) == umap.end()) {
                if (reverse.find(t[i]) != reverse.end()) return false;
                umap[s[i]] = t[i];
                reverse[t[i]] = s[i];
            } else {
                if (t[i] != umap[s[i]]) return false;
            }
        }
        return true;
    }
};